src/intersection.class.ts

```diff
@@ -1,68 +1,91 @@
 //@ts-nocheck
-import { Point } from './point.class';
+import { Point } from "./point.class";
+import { fabric } from '../HEADER';
 
-(function(global) {
-  /* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
-  var fabric = global.fabric || (global.fabric = { });
-  /**
-   * Intersection class
-   * @class fabric.Intersection
-   * @memberOf fabric
-   * @constructor
-   */
-  function Intersection(status) {
+/* Adaptation of work of Kevin Lindsey (kevin@kevlindev.com) */
+
+type IntersectionType = 'Intersection' | 'Coincident' | 'Parallel';
+
+/**
+ * **Assuming `T`, `A`, `B` are points on the same line**,
+ * check if `T` is contained in `[A, B]` by comparing the direction of the vectors from `T` to `A` and `B`
+ * @param T 
+ * @param A 
+ * @param B 
+ * @returns true if `T` is contained
+ */
+const isContainedInInterval = (T: Point, A: Point, B: Point) => {
+  const TA = new Point(T).subtract(A);
+  const TB = new Point(T).subtract(B);
+  return Math.sign(TA.x) !== Math.sign(TB.x) || Math.sign(TA.y) !== Math.sign(TB.y);
+}
+
+export class Intersection {
+
+  points: Point[]
+
+  status?: IntersectionType
+
+  constructor(status?: IntersectionType) {
     this.status = status;
     this.points = [];
   }
 
-  fabric.Intersection = Intersection;
-
-  fabric.Intersection.prototype = /** @lends fabric.Intersection.prototype */ {
-
-    constructor: Intersection,
+  /**
+   * 
+   * @param {Point} point 
+   * @returns 
+   */
+  contains(point) {
+    return this.points.some(p => p.eq(point));
+  }
 
-    /**
-     * Appends a point to intersection
-     * @param {Point} point
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoint: function (point) {
+  /**
+   * Appends a point to intersection
+   * @param {Point} point
+   * @return {Intersection} thisArg
+   * @chainable
+   */
+  appendPoint(point) {
+    if (!this.contains(point)) {
       this.points.push(point);
-      return this;
-    },
-
-    /**
-     * Appends points to intersection
-     * @param {Array} points
-     * @return {fabric.Intersection} thisArg
-     * @chainable
-     */
-    appendPoints: function (points) {
-      this.points = this.points.concat(points);
-      return this;
     }
-  };
+    return this;
+  }
 
   /**
-   * Checks if one line intersects another
-   * TODO: rename in intersectSegmentSegment
+   * Appends points to intersection
+   * @param {Point[]} points
+   * @return {Intersection} thisArg
+   * @chainable
+   */
+  appendPoints(points) {
+    this.points = this.points.concat(points.filter(point => {
+      return !this.contains(point);
```

Comment:
same unique

Reply:
is this hot code? why does it matters? If this is used during render i don't really want to do it.

Reply:
hot code? What's that?

Reply:
How damaging is this loop?
we can make it a map instead but consider that most case points intersection is <=2

Reply:
only 2 polygons that are almost identical will have <= n-1

Reply:
> hot code? What's that?

ahh code used by the dev?
yes, getting intersection points in crucial for ui displaying these points

Reply:
with hot code i mean code we use in tight loops during rendering. Goint to check how many intersections we use

Reply:
Ok is fine, the Intersection is just a fancy container for points, the math to find the point didn't change. I'm good with this.